File: B-example-solutions/obj-c/01/Assignment-01-06/Assignment-01-06/ViewController.m
//
//  ViewController.m
//  Assignment-01-06
//
//  Created by Philip Dow on 8/27/15.
//  Copyright (c) 2015 Phil Dow. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <UITextFieldDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL) textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    
    NSString *text = textField.text;
    NSURL *URL = [NSURL URLWithString:text];
    NSURLRequest *request = [NSURLRequest requestWithURL:URL];
    
    [self.webView loadRequest:request];
    
    return YES;
}

@end

File: B-example-solutions/swift/01/Assignment-01-06/Assignment-01-06/ViewController.swift
//
//  ViewController.swift
//  Assignment-01-06
//
//  Created by Philip Dow on 8/27/15.
//  Copyright (c) 2015 Phil Dow. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITextFieldDelegate {

    @IBOutlet weak var webView: UIWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - Text View Delegate

    func textFieldShouldReturn(textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        
        let text = textField.text
        let URL = NSURL(string: text)
        
        if let URL = URL {
            let request = NSURLRequest(URL: URL)
            webView.loadRequest(request)
        }
        
        return true
    }
}

